Jamfile

```diff
@@ -207,9 +207,9 @@ rule warnings ( properties * )
 	if <toolset>clang in $(properties)
 		|| <toolset>darwin in $(properties)
 	{
-		result += <cflags>-Weverything ;
-		result += <cflags>-Wno-documentation ;
-		result += <cxxflags>-Wno-c++98-compat-pedantic ;
+		result += <cflags>-Wall ;
```

Comment:
I think it would be better if those `cflags` were actually `compileflags` as the first is for C compiler only and the latter is for both C and C++ compiler.

https://www.boost.org/doc/libs/1_78_0/tools/build/doc/html/index.html

I'm unfamiliar with boost build system and I probably should stop here...